### ISSUE TITLE
1 bug and 1 nice to have feature

### ISSUE THREAD

**Issue by vincentneko** [bafolts/terraforming-mars#3]
In a 2 player game, 1 player did not have the 4 card selection to pick at begining  of generation 2 (the other player got it)

This is not a bug but inventor's guild only display card title, it would be great to see full card description in order to make the choice easier.

**Comment by bafolts** [comment]
I pushed a change today that only showed the card title. You have to click it to display the description. I will change it back to show the description as that is simpler.

I wasn't able to recreate the bug with a quick smoke test. The server should send all players to the research phase where they would select project cards, that would be surprising if a player was not taken to the research phase. Did the player that never selected cards during the research phase get to the action phase of the second generation?

**Comment by vincentmoreaudba** [comment]
Thanks for the change.
I did another test with 2 players, I was not able to reproduce the issue. Will give another try with a friend.
On another game, the game was locked, both players were in the "Not your turn to take any actions". I will try to provide additional information next time.

**Comment by vincentneko** [comment]
Thanks for the change.
I did another test with 2 players, I was not able to reproduce the issue. Will give another try with a friend.
On another game, the game was locked, both players were in the "Not your turn to take any actions". I will try to provide additional information next time.

**Comment by bafolts** [comment]
I have ran into this issue as well. Keeping this open until it is clear how to recreate it. It seems to happen to me sometime after having an error when selection an option. Something after an error isn't resetting some state properly is my guess.

**Comment by vincentneko** [comment]
Got another issue at GEN 2, in a 2 player game. Both had the 4 cards research choice.
One was able to select cards, other was struck, clicking the select button did nothing.

Error messages after I CTRL+C process was :

Error running input Unsupported waitingFor
Error processing input from player Unsupported waitingFor
Error running input Unsupported waitingFor
Error processing input from player Unsupported waitingFor
Error running input Unsupported waitingFor
Error processing input from player Unsupported waitingFor
Error running input Unsupported waitingFor
Error processing input from player Unsupported waitingFor
Error running input Unsupported waitingFor
Error processing input from player Unsupported waitingFor
Error running input Unsupported waitingFor
Error processing input from player Unsupported waitingFor

**Comment by bafolts** [comment]
This is most definitely happening after some user is getting the error with their input. That `Unsupported waitingFor` will happen when the server doesn't know how to parse the user input. I will look into how the game state is impacted when an input starts to get processed but never gets completed.

**Comment by vincentneko** [comment]
I tried to undestand how this error is happening but without success.

I'm able to reproduce it doing the following:

- Create a 2 player game
- Choose a corp and 2 cards for each player
- Pass for each player
- Player one is able to select cards during research phase
- Player 2 is struck when submiting research phase cards

Error message is: 
Error running input Unsupported waitingFor
Error processing input from player Unsupported waitingFor

**Comment by bafolts** [comment]
Your steps to recreate this issue were helpful. I spotted at least two issues with the steps you included to reproduce. I think with the fix a lot of these type of issues should go away. We can create new issues if we run into more and include the steps to reproduce. I also displayed all card information by default again until we come up with a better solution that always requiring the card to be clicked to see the card information.
